src/libYARP_dev/src/IVisualParamsImpl.cpp

```diff
@@ -202,12 +208,11 @@ bool Implement_RgbVisualParams_Parser::respond(const yarp::os::Bottle& cmd, yarp
                     ret = iRgbVisual->getRgbIntrinsicParam(params);
                     if(ret)
                     {
-                        yarp::os::Bottle params_b;
                         response.addVocab(VOCAB_RGB_VISUAL_PARAMS);
                         response.addVocab(VOCAB_INTRINSIC_PARAM);
                         response.addVocab(VOCAB_IS);
-                        ret &= Property::copyPortable(params, params_b);  // will it really work??
-                        response.append(params_b);
+                        yarp::os::Bottle& tmp=response.addList();
+                        ret &= Property::copyPortable(params, tmp);
```

Comment:
I was quite confused by the use of bitwise operation on bool operands. Even if it should work [1] have you considered changing it for readability? 

[1] : http://timmurphy.org/2012/05/08/bool-bitwise-and-equals-bool-in-c/